app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+  before_action :user_authenticated, only: %i[index show]
+  include UserHelper
+
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    respond_to do |format|
+      if @user.save
+        session[:name] = params[:name]
+        format.html { redirect_to users_path, notice: 'User was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def show
```

Comment:
The UsersController#show:

- [ ] lists events created by user, i.e. @created_events =  current_user.events